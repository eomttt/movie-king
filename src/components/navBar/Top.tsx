import React from 'react';

import styled from 'styled-components';

import {
  mobileOffSet,
  defaultBackgroundColor,
  topNavBarZIndex,
  topNavBarHeight,
} from '@styles/common';

import { LOGO_ICON } from '@constants/common';

const Container = styled.div`
  width: 100%;
  height: ${topNavBarHeight}px;
  max-width: ${mobileOffSet}px;
  position: fixed;
  top: 0;
  z-index: ${topNavBarZIndex};
  background-color: ${defaultBackgroundColor};
`;

const Content = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  & img {
    width: 39px;
    margin: auto;
  }
`;

const Top = () => (
  <Container>
    <Content>
      <img src={LOGO_ICON} alt="Logo" />
    </Content>
  </Container>
);

export default Top;
