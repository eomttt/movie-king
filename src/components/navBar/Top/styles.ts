import {
  navBarColor,
  mobileOffSet,
  topNavBarHeight,
  topNavBarZIndex,
} from 'styles/common';
import styled from '@emotion/styled';

export const Container = styled.div`
  width: 100%;
  height: ${topNavBarHeight}px;
  max-width: ${mobileOffSet}px;
  position: fixed;
  top: 0;
  z-index: ${topNavBarZIndex};
  background-color: ${navBarColor};
`;

export const Content = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  & img {
    width: 39px;
    margin: auto;
  }
`;
