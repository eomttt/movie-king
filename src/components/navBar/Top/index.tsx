import * as Styles from './styles';

const Top = () => (
  <Styles.Container>
    <Styles.Content>
      <p>FLICK</p>
    </Styles.Content>
  </Styles.Container>
);

export default Top;
