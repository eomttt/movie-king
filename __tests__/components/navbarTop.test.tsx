import { render } from '@testing-library/react';
import Top from 'components/navBar/Top';

describe('Navbar top test', () => {
  it('Render correctly normal', () => {
    const top = render(
      <Top />,
    );
    expect(top.container).toMatchSnapshot();
  });
});
