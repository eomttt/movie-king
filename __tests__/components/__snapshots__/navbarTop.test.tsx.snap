// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar top test Render correctly normal 1`] = `
<div>
  <div
    class="css-1ctkiov-Top_Container e7nygkw0"
  >
    <div
      class="css-inu4ig-Top_Content e7nygkw1"
    >
      <img
        alt="Logo"
        src="https://d2mgzmtdeipcjp.cloudfront.net/files/upload/15680246797940.svg"
      />
    </div>
  </div>
</div>
`;
